.github/workflows/test_libs.yml

```diff
@@ -31,14 +33,37 @@ jobs:
             - 'libs/tktokenswap/**'
           tkwsm:
             - 'libs/tkwsm/**'
+          gh_actions:
+            - '.github/workflows/test_libs.yml'
+  set_libs_matrix:
+    name: Set the libs strategy matrix
+    needs: changes
+    runs-on: ubuntu-22.04
+    outputs:
+      libs: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Set LIBS_TO_TEST to all the libraries
+        # Test all libraries if the test_libs.yml file was modified
+        if: ${{ contains(needs.changes.outputs.libs, 'gh_actions') }}
+        run: |
+          echo "LIBS_TO_TEST=${{ toJSON(env.ALL_LIBS) }}" >> $GITHUB_ENV
+      - name: Set LIBS_TO_TEST to the modified libraries only
+        if: ${{ ! contains(needs.changes.outputs.libs, 'gh_actions') }}
+        run: |
+          echo "LIBS_TO_TEST=${{ toJSON(needs.changes.outputs.libs) }}" >> $GITHUB_ENV
+      - name: Set libs matrix
+        id: set-matrix
+        run: |
+          echo "::set-output name=matrix::$LIBS_TO_TEST"
   test_libraries:
     name: test library
-    needs: changes
-    if: ${{ needs.changes.outputs.libs != '[]' && needs.changes.outputs.libs != '' }}
+    needs: set_libs_matrix
+    if: ${{ needs.set_libs_matrix.outputs.libs != '[]' && needs.set_libs_matrix.outputs.libs != '' }}
     strategy:
+      fail-fast: false
```

Comment:
We want the workflow to fail if any of these jobs fails. Is this still the case with this set to `false`?

Reply:
Yes, that's still the case. The `fail-fast: false` only means that any matrix jobs still running will not be canceled if another job fails. At the end, the workflow is reported as failed if any of those jobs failed.